modules/kibana/src/javaRestTest/java/org/elasticsearch/kibana/KibanaSystemIndexIT.java

```diff
@@ -27,25 +29,43 @@
 import org.elasticsearch.test.rest.ESRestTestCase;
 
 import java.io.IOException;
+import java.util.Arrays;
 import java.util.Map;
 
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.is;
 
 public class KibanaSystemIndexIT extends ESRestTestCase {
 
+    private final String indexName;
+
+    public KibanaSystemIndexIT(@Name("indexName") String indexName) {
+        this.indexName = indexName;
+    }
+
+    @ParametersFactory
```

Comment:
It will run the test suite four times, once with each parameter. I believe we could adjust this method to return a random subset of the options if we wanted to cut the test time for this suite.

We could also, for example, add a boolean flag `isKibanaSystemIndex` as a second parameter and have the simpler tests verify that invalid patterns don't work with this API, while using `assumeTrue` to skip the more complex tests for those parameters.

Reply:
Cool, that's what I expected - I don't think we want to run it on a random subset, I'd rather have the coverage every run. 

I think for this case it's fine, but we probably want to be careful doing this for integration tests depending on whether our infra spins up 4 clusters to run the suite 4 times or something like that which might cause it to take extra time vs. just handling the parameters manually. I don't know enough to answer that off the top of my head, but something to look into. (Don't block this PR on that though)